### PR TITLE
case insensitive text filter

### PR DESCRIPTION
**Pull request by makenova**
For issue #189

### CODE REVIEW

File: packages/react-bootstrap-table2-filter/src/filter.js [hunk 1]
```diff
@@ -16,7 +16,8 @@ export const filterByText = _ => (
     if (comparator === EQ) {
       return cellStr === filterVal;
     }
-    return cellStr.indexOf(filterVal) > -1;
+    const re = RegExp(filterVal, 'i');
```
Comment (AllenFang) [inline packages/react-bootstrap-table2-filter/src/filter.js]: is that danger? for example `\test\` or `\\test\\`, maybe throw some err?

**Comment by makenova** [comment]
Making it configureable is a great idea!
Pls show me how you’d like it changed.

File: packages/react-bootstrap-table2-filter/src/filter.js (hunk 1 above)
Comment (makenova) [reply packages/react-bootstrap-table2-filter/src/filter.js]: Like if the `filterVal` is `\test\`?
I’ll do some testing to see if that can cause an error.

**Comment by AllenFang** [comment]
@makenova I think we can allow the text filter to have caseInsensitive via `textFilter` props.
Currently, text and select filter allow user to configure some thing, check [this](https://github.com/react-bootstrap-table/react-bootstrap-table2/blob/master/packages/react-bootstrap-table2-example/examples/column-filter/custom-text-filter.js#L46-L52) example, I think you can have the same way to do it.

add a new `caseSensitive` props for text filter
then on [here](https://github.com/react-bootstrap-table/react-bootstrap-table2/blob/master/packages/react-bootstrap-table2-filter/src/wrapper.js#L53) get `caseSensitive` props, for example:

```js
```

**Comment by makenova** [comment]
I've updated it with `caseSensitive` props and moved away for using `RegExp`.

**Comment by AllenFang** [comment]
@makenova  thanks your contribution 👍

**Comment by AllenFang** [comment]
https://github.com/react-bootstrap-table/react-bootstrap-table2/pull/201

**Pull request merged**